pkg/iac/terraform/resource_block.go

```diff
@@ -171,3 +165,29 @@ func renderMap(val map[string]any) string {
 	result = fmt.Sprintf("%s}", result)
 	return result
 }
+
+func escapeSpecialSequences(input string) string {
+	var sb strings.Builder
+	sb.Grow(len(input))
+	for i, r := range input {
+		if r == '$' || r == '%' {
+			sb.WriteRune(r)
+			remain := input[i+1:]
+
+			// it's not a special sequence
+			if remain == "" || remain[0] != '{' {
+				continue
+			}
+
+			// sequence is already escaped
+			if i > 0 && rune(input[i-1]) == r {
+				continue
+			}
+
+			sb.WriteRune(r)
+		} else {
+			sb.WriteRune(r)
+		}
+	}
+	return sb.String()
+}
```

Comment:
I wonder if hashicorp already has this logic somewhere that we could borrow?

Reply:
internal only

Reply:
ah bummer

Reply:
could you cite the source in that case?

Reply:
There's not much left of the source https://github.com/hashicorp/hcl/blob/main/hclwrite/generate.go#L367-L373